cpp/include/cudf/utilities/logger.hpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+// If using GCC, temporary workaround for older libcudacxx defining _LIBCPP_VERSION
+// undefine it before including spdlog, due to fmtlib checking if it is defined
+// TODO: remove once libcudacxx is on Github and RAPIDS depends on it
+#ifdef __GNUG__
+#undef _LIBCPP_VERSION
+#endif
+#include <spdlog/sinks/basic_file_sink.h>
+#include <spdlog/spdlog.h>
```

Comment:
Is it possible to not require this third-party header in one of our public headers?
This becomes a frustrating sequence of dependencies when writing code to use libcudf APIs.
For example, adjusting my `-I` parameter to include headers that I don't use directly.
Would it be possible to code this using the [pimpl](https://en.cppreference.com/w/cpp/language/pimpl) approach?
This moves the third-party dependencies to a .cpp file instead of the public header.

Reply:
looking into this

Reply:
resolving based on offline discussion; removing the `spdlog` dependency would required whole new cudf logging APIs